plugins/module_utils/network/sonic/config/mfa/mfa.py

```diff
@@ -0,0 +1,759 @@
+    to_request,
+    edit_config
+)
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.utils.utils import (
+    update_states,
+    get_diff,
+    remove_empties,
+    remove_empties_from_list
+)
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.utils.formatted_diff_utils import (
+    __DELETE_CONFIG_IF_NO_SUBCONFIG,
+    __DELETE_LEAFS_OR_CONFIG_IF_NO_NON_KEY_LEAF,
+    get_new_config,
+    get_formatted_config_diff
+)
+from ansible.module_utils.connection import ConnectionError
+
+PATCH = 'patch'
+DELETE = 'delete'
+
+TEST_KEYS = [
+    {'rsa_servers': {'hostname': ''}}
+]
+
+TEST_KEYS_formatted_diff = [
+    {'config': {'__delete_op': __DELETE_LEAFS_OR_CONFIG_IF_NO_NON_KEY_LEAF}},
+    {'rsa_servers': {'hostname': '', '__delete_op': __DELETE_CONFIG_IF_NO_SUBCONFIG}},
+]
+
+MFA_PATH = 'data/openconfig-mfa:mfa'
+
+
+class Mfa(ConfigBase):
+    """
+    The sonic_mfa class
+    """
+
+    gather_subset = [
+        '!all',
+        '!min',
+    ]
+
+    gather_network_resources = [
+        'mfa',
+    ]
+
+    def __init__(self, module):
+        super(Mfa, self).__init__(module)
+
+    def get_mfa_facts(self):
+        """ Get the 'facts' (the current configuration)
+
+        :rtype: A dictionary
+        :returns: The current configuration as a dictionary
+        """
+        facts, _warnings = Facts(self._module).get_facts(self.gather_subset, self.gather_network_resources)
+        mfa_facts = facts['ansible_network_resources'].get('mfa')
+        if not mfa_facts:
+            return []
+        return mfa_facts
+
+    def execute_module(self):
+        """ Execute the module
+
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        result = {'changed': False}
+        warnings = []
+        commands = []
+
+        existing_mfa_facts = self.get_mfa_facts()
+        commands, requests = self.set_config(existing_mfa_facts)
+        if commands and len(requests) > 0:
+            if not self._module.check_mode:
+                try:
+                    edit_config(self._module, to_request(self._module, requests))
+                except ConnectionError as exc:
+                    self._module.fail_json(msg=str(exc), code=exc.code)
+            result['changed'] = True
+        result['commands'] = commands
+
+        changed_mfa_facts = self.get_mfa_facts()
+
+        result['before'] = existing_mfa_facts
+        if result['changed']:
+            result['after'] = changed_mfa_facts
+
+        new_config = changed_mfa_facts
+        old_config = existing_mfa_facts
+        if self._module.check_mode:
+            result.pop('after', None)
+            new_commands = deepcopy(commands)
+            new_config = get_new_config(new_commands, old_config, TEST_KEYS_formatted_diff)
+            result['after(generated)'] = self._post_process_generated_output(new_config)
```

Comment:
Please remove duplicate facts retrieval in check mode.
```suggestion
        result['before'] = existing_mfa_facts
        old_config = existing_mfa_facts

        if self._module.check_mode:
            new_commands = deepcopy(commands)
            new_config = get_new_config(new_commands, old_config, TEST_KEYS_formatted_diff)
            result['after(generated)'] = self._post_process_generated_output(new_config)
        else:
            new_config = self.get_mfa_facts()
            if result['changed']:
                result['after'] = new_config
```